camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+        fs.delete(tmpPath, true);
+        throw new RuntimeException("Error: cannot rename " + tmpPath + " to " + outputPath);
+      }
+      deleteOldPath(oldPath);
+    }
+
+    private void deleteOldPath(Path oldPath) throws IOException {
+      if (oldPath != null && fs.exists(oldPath)) {
+        LOG.info("Deleting " + oldPath);
+        fs.delete(oldPath, true);
+      }
+    }
+
+    private void moveExistingContentInOutputPathToOldPath(Path oldPath) throws IOException {
+      LOG.info("Path " + outputPath + " exists. Overwriting.");
+      if (!fs.rename(outputPath, oldPath)) {
+        fs.delete(tmpPath, true);
+        throw new RuntimeException("Error: cannot rename " + outputPath + " to " + oldPath);
+      }
+    }
+
+    private void setNumOfReducersAndSplitSizes() throws IOException {
+      long inputSize = getInputSize();
+
+      int maxFiles = job.getConfiguration().getInt(MAX_FILES, DEFAULT_MAX_FILES);
+      int numTasks = Math.min((int) (inputSize / targetFileSize) + 1, maxFiles);
+
+      if (job.getNumReduceTasks() != 0) {
+        determineAndSetNumOfReducers(numTasks);
+      } else {
+        setSplitSizes(inputSize / numTasks);
+      }
+    }
+
+    private void submitMrJob() throws IOException, InterruptedException, ClassNotFoundException {
+      CamusHourlySweeper.this.metrics.mrSubmitTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+      job.submit();
+      runningJobs.add(job);
+
+      CamusHourlySweeper.this.metrics.mrStartRunningTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
+
+      LOG.info("job running for: " + job.getConfiguration().get(TOPIC_AND_HOUR) + ", url: " + job.getTrackingURL());
+      job.waitForCompletion(false);
+      if (!job.isSuccessful()) {
+        throw new RuntimeException("hadoop job failed.");
+      }
+    }
+
+    private void setSplitSizes(long targetSplitSize) {
+      LOG.info("Setting target split size " + targetSplitSize);
+      job.getConfiguration().setLong(MAPRED_MAX_SPLIT_SIZE, targetSplitSize);
+      job.getConfiguration().setLong(MAPRED_MIN_SPLIT_SIZE, targetSplitSize);
+    }
+
+    private void determineAndSetNumOfReducers(int numTasks) {
+      int numReducers;
+      if (job.getConfiguration().get(REDUCER_COUNT) != null) {
+        numReducers = job.getConfiguration().getInt(REDUCER_COUNT, DEFAULT_REDUCER_COUNT);
+      } else {
+        numReducers = numTasks;
+      }
+      job.setNumReduceTasks(numReducers);
+    }
+
+    private long getInputSize() throws IOException {
+      long inputSize = 0;
+      for (Path p : inputPaths) {
+        LOG.info("inputPath: " + p.toString() + ", size=" + fs.getContentSummary(p).getLength());
+        inputSize += fs.getContentSummary(p).getLength();
+      }
+      return inputSize;
+    }
+
+    private void addInputAndOutputPathsToFileInputFormat() throws IOException {
+      for (Path path : inputPaths) {
+        FileInputFormat.addInputPath(job, path);
```

Comment:
There's also `FileInputFormat.setInputPaths()` that allows you to set a collection of paths.